.github/workflows/node.js.yml

```diff
@@ -0,0 +1,45 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Test
+
+on:
+  push:
+    branches: []
+    tags: ["*"]
+
+jobs:
+  directories:
+    runs-on: ubuntu-latest
+    outputs:
+      dir: ${{ steps.dir.outputs.dir }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: dir
+        run: echo "::set-output name=dir::$(ls -d packages/*/ | jq --raw-input --slurp --compact-output 'split("\n")[:-1]')"
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: "yarn"
+      - run: yarn
+      - run: |
+          cd packages/sign
+          yarn build
+  testing:
+    runs-on: ubuntu-latest
+    needs: [directories, build]
+    strategy:
+      matrix:
+        dir: ${{fromJson(needs.directories.outputs.dir)}}
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          cd ./${{ matrix.dir }}
```

Comment:
Use [`working-directory`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_iddefaultsrun).

Reply:
Um, actually I want to talk about this with you, that I'm currently contemplating about our directory structure of source codes. 

I think @planetarium/sign should be treated differently than account implementations, like, yes they are all packages treated under monorepo but I think these should be bundled by it's usage. for example, 'account's should be bundled in account directory and tested independently from 'sign', since these accounts are all binded by account interface in 'sign', and account implementations are going to be more numerous from now and they have pretty much simillar inter-structure, but on the other hand, what name should we put on 'sign' directory? 'API' seems too broad, 'signer' seems to narrow, 'interface' is also too ambiguous. 

Why am I writing about this long directory-path-nonsense? because I'm considering automating (and separating) test execution by the separate script and with a little adjustment on vitest config, executing all test on repository root directory so it doesn't need to be depended by it's directory path. whaddya think?